internal/services/storage/storage_share_resource.go

```diff
@@ -143,239 +150,424 @@ func resourceStorageShare() *pluginsdk.Resource {
+			log.Printf("[DEBUG] %q was not found, removing from state", *id)
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("retrieving %q: %v", *id, err)
 	}
-	if props == nil {
-		log.Printf("[DEBUG] File Share %q was not found in %s - assuming removed & removing from state", id.ShareName, account.StorageAccountId)
-		d.SetId("")
-		return nil
+
+	d.Set("storage_account_id", commonids.NewStorageAccountID(id.SubscriptionId, id.ResourceGroupName, id.StorageAccountName).ID())
+
+	if model := existing.Model; model != nil {
+		d.Set("name", id.ShareName)
+		if props := model.Properties; props != nil {
+			d.Set("quota", props.ShareQuota)
+			// Resource Manager treats nil and "SMB" as the same and we may not get a full response here
+			enabledProtocols := fileshares.EnabledProtocolsSMB
+			if props.EnabledProtocols != nil {
+				enabledProtocols = *props.EnabledProtocols
+			}
+			d.Set("enabled_protocol", string(enabledProtocols))
+			d.Set("access_tier", string(pointer.From(props.AccessTier)))
+			d.Set("acl", flattenStorageShareACLs(pointer.From(props.SignedIdentifiers)))
+			d.Set("metadata", FlattenMetaData(pointer.From(props.Metadata)))
+		}
 	}
 
-	d.Set("name", id.ShareName)
-	d.Set("storage_account_name", id.AccountId.AccountName)
-	d.Set("quota", props.QuotaGB)
-	d.Set("url", id.ID())
-	d.Set("enabled_protocol", string(props.EnabledProtocol))
+	d.Set("resource_manager_id", id.ID())
 
-	accessTier := ""
-	if props.AccessTier != nil {
-		accessTier = string(*props.AccessTier)
+	account, err := storageClient.FindAccount(ctx, subscriptionId, id.StorageAccountName)
+	if err != nil {
+		return fmt.Errorf("retrieving Account %q for Share %q: %v", id.StorageAccountName, id.ShareName, err)
+	}
+	if account == nil {
+		return fmt.Errorf("locating Storage Account %q", id.StorageAccountName)
 	}
-	d.Set("access_tier", accessTier)
 
-	if err := d.Set("acl", flattenStorageShareACLs(props.ACLs)); err != nil {
-		return fmt.Errorf("flattening `acl`: %+v", err)
+	// Determine the file endpoint, so we can build a data plane ID
+	endpoint, err := account.DataPlaneEndpoint(client.EndpointTypeFile)
+	if err != nil {
+		return fmt.Errorf("determining File endpoint: %v", err)
 	}
 
-	if err := d.Set("metadata", FlattenMetaData(props.MetaData)); err != nil {
-		return fmt.Errorf("flattening `metadata`: %+v", err)
+	// Parse the file endpoint as a data plane account ID
+	accountId, err := accounts.ParseAccountID(*endpoint, storageClient.StorageDomainSuffix)
+	if err != nil {
+		return fmt.Errorf("parsing Account ID: %v", err)
 	}
 
-	resourceManagerId := parse.NewStorageShareResourceManagerID(account.StorageAccountId.SubscriptionId, account.StorageAccountId.ResourceGroupName, account.StorageAccountId.StorageAccountName, "default", id.ShareName)
-	d.Set("resource_manager_id", resourceManagerId.ID())
+	// TODO Giovanni Deprecation - Port the storage data plane ID helper(s) from Giovanni to go-azure-sdk so we can remove this
```

Comment:
Which helpers specifically, the ones above? all of them?

Reply:
Ultimately yes, we need to deprecate the use of Giovanni completely, but that work is outside the scope of this PR.